docs/adr/ADR-009-polyfill-argument-null-exception.md

```
@@ -0,0 +1,152 @@
+
+## Future Work
+
+- Monitor C# extension members feature stability in future SDK releases
+- Consider consolidating polyfill files into a shared source package if pattern proves stable
+- Evaluate if `[Embedded]` attribute becomes unnecessary in future Roslyn versions
+
+## References
+
+- Andrew Lock: "Behind the implementation of the [Embedded] attribute in .NET 10" - <https://andrewlock.net/behind-the-implementation-of-the-embedded-attribute-in-net-10/>
+- SimonCropp/Polyfill `ArgumentNullExceptionPolyfill.cs` (commit e78ac432695270075490e9fbee211e25e4fccb70)
+- C# Extension Members (C# 14 proposal): https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-14.0/extensions
```

markdownlint-cli2 (0.18.1)

Bare URL used

(MD034, no-bare-urls)